pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -92,6 +100,23 @@ def is_informative(self):
         """Return True is the transform does not need to be executed."""
         return self._is_informative
 
+    def _default_qnode_transform(self, qnode, targs, tkwargs):
+        """
+        The default method that takes in a QNode and returns another QNode
+        with the transform applied.
+        """
+        if qnode.transform_program.is_empty():
```

Comment:
Out of scope for this PR, but why only if its empty?

Reply:
Oh, I'm not sure. It seems to me like it should be copied in every scenario to avoid any side effects.

Reply:
Most cases users would deepcopy the qnode too many times, but it is true that it can have some side effect. You can remove the condition.